### DISCUSSION TITLE
Foreground cache에서 왜 Thread를 캐싱할까?

### DISCUSSION THREAD

**Discussion by CODe5753** [Well-Founded-Confidence/RealMySQL#7] [Q&A] [answered]
보통 캐싱은 어떤 자주 사용되는 데이터를 빠르게 가져오기 위해 사용되는 것으로 알고 있는데,
Foreground cache 설명에서 Thread를 캐싱한다는 내용을 보았습니다.
Thread 캐싱은 조금 생소한데 왜 Thread를 캐싱하는걸까요?

**Comment by Soyflat** [comment]
정답이 아닐 수 있습니다
Foreground cache는 MySQL의 InnoDB 스토리지 엔진에서 사용되는 캐시 메모리로, InnoDB 버퍼 풀과는 별개로 동작합니다.
Foreground cache는 쿼리 결과를 캐시하는 것뿐만 아니라, 쿼리 실행에 필요한 다른 요소들도 캐시합니다. 그 중 하나가 스레드(Thread)입니다. Foreground cache는 스레드를 캐시함으로써, 새로운 스레드를 만드는 데 드는 비용을 줄이고, 이미 존재하는 스레드를 재사용함으로써, 스레드 생성 및 소멸에 따른 오버헤드를 감소시킵니다. 이를 통해 더 빠른 쿼리 실행 속도와 더 적은 시스템 리소스 소모를 달성할 수 있습니다.

따라서, Foreground cache는 스레드 캐싱을 통해 MySQL의 쿼리 실행 속도를 향상시키는 데 기여합니다.

**Comment by menthamin** [comment, marked as answer]
MVCC
- 버전체킹 I/O 발생
- 버전체킹 시 스레드의 정보가 활용됨
- 스레드를 지속 생성 / 삭제 하면 버전 체킹 정보가 소실됨
